varlink/connection.go

```diff
@@ -238,6 +245,24 @@ func (c *Connection) GetInfo(ctx context.Context, vendor *string, product *strin
 	return nil
 }
 
+// Upgrade attempts to upgrade the connection using the provided method and parameters.
+// If successful, the connection cannot be reused later, and must be closed.
+func (c *Connection) Upgrade(ctx context.Context, method string, parameters interface{}) (func(context.Context, interface{}) (uint64, ReadWriterContext, error), error) {
```

Comment:
Isn't this actually a call with upgrade flag forced on? I wander if the name should be more explicit, like "CallWithUpgrade" or "SendAndUpgrade" or something like this

Reply:
I think it's consistent enough with the rest of the names, and the documentation covers the use cases without needing to disclose implementation details.

Reply:
Works for me, thanks!